neo/ProtocolSettings.cs

```diff
@@ -11,12 +12,31 @@ public class ProtocolSettings
         public string[] SeedList { get; }
         public uint SecondsPerBlock { get; }
 
-        public static ProtocolSettings Default { get; }
+        static ProtocolSettings _default;
 
-        static ProtocolSettings()
+        static bool UpdateDefault(IConfiguration configuration)
         {
-            IConfigurationSection section = new ConfigurationBuilder().AddJsonFile("protocol.json", true).Build().GetSection("ProtocolConfiguration");
-            Default = new ProtocolSettings(section);
+            var settings = new ProtocolSettings(configuration.GetSection("ProtocolConfiguration"));
+            return null == Interlocked.CompareExchange(ref _default, settings, null);
```

Comment:
When it `return true`? only when change the first time?

Reply:
Yes, UpdateDefault will only return true the first time it was called. [CompareExchange](https://docs.microsoft.com/en-us/dotnet/api/system.threading.interlocked.compareexchange) atomically compares _default to null and stores settings in _default if and only if the comparison succeeds. 

I agree with @erikzhang - a node can't switch networks once a node is up and running. This code ensures _default's value can only be written once - even there are multiple threads in the process accessing ProtocolSettings.Default.